tests/expectations/metrics/test_map_metric.py

```diff
@@ -1148,3 +1148,694 @@ def test_sqlite_single_column_summary_result_format(
+        kwargs={
+            "column": "animals",
+            "value_set": ["cat", "fish", "dog"],
+            "result_format": {
+                "result_format": "SUMMARY",  # SUMMARY will include partial_unexpected* values only
+                "unexpected_index_column_names": ["pk_1", "pk_2"],  # Multiple columns
+                "partial_unexpected_count": 1,
+            },
+        },
+    )
+
+    expectation = ExpectColumnValuesToBeInSet(expectationConfiguration)
+    sqlite_path = file_relative_path(__file__, "../../test_sets/metrics_test.db")
+    connection_string = f"sqlite:///{sqlite_path}"
+    engine = SqlAlchemyExecutionEngine(connection_string=connection_string)
+    execution_engine = engine
+    my_data_connector: ConfiguredAssetSqlDataConnector = (
+        ConfiguredAssetSqlDataConnector(
+            name="my_sql_data_connector",
+            datasource_name="my_test_datasource",
+            execution_engine=execution_engine,
+            assets={
+                "my_asset": {
+                    "table_name": "animal_names",
+                },
+            },
+        )
+    )
+    batch_definition_list = (
+        my_data_connector.get_batch_definition_list_from_batch_request(
+            batch_request=BatchRequest(
+                datasource_name="my_test_datasource",
+                data_connector_name="my_sql_data_connector",
+                data_asset_name="my_asset",
+            )
+        )
+    )
+    assert len(batch_definition_list) == 1
+    batch_spec: SqlAlchemyDatasourceBatchSpec = my_data_connector.build_batch_spec(
+        batch_definition=batch_definition_list[0]
+    )
+    batch_data, batch_markers = execution_engine.get_batch_data_and_markers(
+        batch_spec=batch_spec
+    )
+    batch = Batch(data=batch_data)
+    validator = Validator(execution_engine, batches=[batch])
+    result = expectation.validate(validator)
+    assert convert_to_json_serializable(result.result) == {
+        "element_count": 6,
+        "missing_count": 0,
+        "missing_percent": 0.0,
+        "partial_unexpected_counts": [{"count": 1, "value": "giraffe"}],
+        "partial_unexpected_index_list": [{"pk_1": 3, "pk_2": "three"}],
+        "partial_unexpected_list": ["giraffe"],
+        "unexpected_count": 3,
+        "unexpected_percent": 50.0,
+        "unexpected_percent_nonmissing": 50.0,
+        "unexpected_percent_total": 50.0,
+    }
+
+
+def test_sql_multiple_unexpected_index_column_names_basic_result_format(
+    sa, sqlite_table_for_unexpected_rows_with_index
+):
+    expectationConfiguration = ExpectationConfiguration(
+        expectation_type="expect_column_values_to_be_in_set",
+        kwargs={
+            "column": "animals",
+            "value_set": ["cat", "fish", "dog"],
+            "result_format": {
+                "result_format": "BASIC",  # SUMMARY will include partial_unexpected_list only, which means unexpected_index_column_names will have no effect
```

Comment:
This would be worth discussing with @tal, I would have expected a partial_unexpected_index_list to be included if unexpected_index_column_names are specified. Is it not because of the result_format being BASIC?

Reply:
By definition you can't have an expected element in the unexpected list.

Given the nature of this I'd also expect the `result_format` to be `FOURTRAN` instead of `BASIC`.

All in i think this is completely out of bounds and you should be ashamed.